src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeServiceTest.kt

```diff
@@ -23,24 +23,29 @@ class BarcodeServiceTest {
   private val barcodeRepository = mock<BarcodeRepository>()
   private val barcodeEventService = mock<BarcodeEventService>()
   private val barcodeGeneratorService = mock<BarcodeGeneratorService>()
-  private val barcodeService = BarcodeService(barcodeRepository, barcodeEventService, barcodeGeneratorService)
+  private val barcodeRecipientService = mock<BarcodeRecipientService>()
+  private val barcodeService = BarcodeService(barcodeRepository, barcodeEventService, barcodeGeneratorService, barcodeRecipientService)
 
   @Nested
   inner class CreateBarcode {
     @Test
-    fun `should create barcode and created event if ok`() {
+    fun `should create barcode and created event and barcode recipient if ok`() {
```

Comment:
I'm not proposing a unit test here that tests the service whe`n a null `CreateBarcodeRequest` is passed in, as that it a short lived situation.
It is however covered (temporarily) in the integration test `BarcodeResourceTest`